setup.py

```diff
@@ -20,14 +20,72 @@
         sys.exit(1)
 
 
+requirements = {
+    'install': [
+        'filelock',
+        'numpy>=1.9.0',
+        'protobuf>=3.0.0',
+        'six>=1.9.0',
+    ],
+    'cuda': [
+        'cupy==4.0.0b1',
+    ],
+    'stylecheck': [
+        'hacking',
+        'autopep8',
+    ],
+    'test': [
+        'pytest',
+        'mock',
+    ],
+    'doctest': [
+        'matplotlib',
+        'theano',
+    ],
+    'docs': [
+        'sphinx_rtd_theme',
+    ],
+    'travis': [
+        '-rstylecheck',
+        '-rtest',
+        'pytest-timeout',
+        'pytest-cov',
+        'theano',
+        'h5py',
+        'pillow',
+    ],
+    'appveyor': [
+        '-rstylecheck',
+        '-rtest',
+        'pytest-timeout',
+        'pytest-cov',
+    ],
+}
+
+
+def reduce_requirements(key):
+    # Resolve recursive requirements notation (-r)
+    reqs = requirements[key]
+    resolved_reqs = []
+    for req in reqs:
+        if req.startswith('-r'):
```

Comment:
I feel `-r` prefix is a bit non-intuitive.
How about using `@`? I borrowed this idea from yum/dnf package manager (e.g., `yum install @x11` to install all packages in  "x11" group)

Reply:
I borrowed `-r` from [requirements.txt](https://pip.pypa.io/en/stable/reference/pip_install/#requirements-file-format).
I think it's better to imitate it (perhaps we can insert a space?).

Reply:
I see. Then, inserting a space makes it more understandable I think.

Reply:
OK.